File: STORE/backend/app/models/attendance.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from app.models.base import TimestampedDocument


AttendanceStatus = Literal["PRESENT", "LATE", "ABSENT", "LEAVE"]


class AttendanceDocument(TimestampedDocument):
    businessId: str
    userId: str
    storeId: str
    clockIn: datetime | None = None
    clockOut: datetime | None = None
    date: str
    status: AttendanceStatus = "PRESENT"

File: STORE/mobile/src/api/endpoints/attendance.ts
import { apiClient } from '@/api/client';

export type AttendanceStatus = 'PRESENT' | 'LATE' | 'ABSENT' | 'LEAVE';

export interface AttendanceRecord {
  id: string;
  userId: string;
  businessId?: string;
  storeId?: string | null;
  date: string;
  status: AttendanceStatus;
  clockIn?: string | null;
  clockOut?: string | null;
  note?: string | null;
  createdAt?: string;
  updatedAt?: string;
}

export async function getAttendanceToday(): Promise<AttendanceRecord> {
  return apiClient.get('/api/v1/attendance/today');
}

export async function clockIn(payload: { storeId?: string | null } = {}): Promise<AttendanceRecord> {
  return apiClient.post('/api/v1/attendance/clock-in', payload);
}

export async function clockOut(): Promise<AttendanceRecord> {
  return apiClient.post('/api/v1/attendance/clock-out', {});
}